### ISSUE TITLE
How to mirro into an organization instead of an user

### ISSUE THREAD

**Issue by XargonWan** [jaedle/mirror-to-gitea#20]
Greetings,

first of all, thank you for this software, it's really making the life easier!

I am using this mirrot to migrate my RetroDECK organization inside my own forgejo instance, however I noticed that in forgejo I can only generate tokens for users and not for organization so the destination of the mirroring is the user, I wish them to be migrated inside an organization named RetroDECK.
Is it possible?

Thanks in advance.

**Comment by opensourcer2** [comment]
+1

**Comment by XargonWan** [comment]
> Hey 👋
> 
> I am having a look if this would be easily possible.

Thanks a lot, sorry for the late reply!

**Comment by CrazyWolf13** [comment]
I'd love to see this feature as well, thanks for checking, more versatil output settings would be awesome!

**Comment by lucaspalomodevelop** [comment]
+1

**Comment by outbackdingo** [comment]
++1

**Comment by arunavo4** [comment]
Hey I made something similar for the starred repos in my github account where all starred repos are moved into a organisation if the user wants. There is a PR out for it

**Comment by arunavo4** [comment]
@XargonWan

Yes, it's absolutely possible to mirror repositories to a specific organization in Forgejo, even though you can only generate tokens for users. The mirror-to-gitea tool has a feature specifically for this purpose.

You can use the `GITEA_ORGANIZATION` environment variable to specify the target organization in Forgejo where all repositories should be mirrored. Here's a Docker command that should work for your use case:

```bash
docker run -it --rm \
  -e GITHUB_USERNAME="your-github-username" \
  -e GITEA_URL="https://your-forgejo-instance-url" \
  -e GITEA_TOKEN="your-user-token-from-forgejo" \
  -e GITHUB_TOKEN="your-github-token" \
  -e GITEA_ORGANIZATION="RetroDECK" \
  -e MIRROR_ORGANIZATIONS="true" \
  -e DRY_RUN="true" \
  arunavo4/mirror-to-gitea:latest
```

This command will:
1. Connect to GitHub using your credentials
2. Fetch repositories from the RetroDECK organization on GitHub
3. Mirror them to the RetroDECK organization in your Forgejo instance
4. Run in dry-run mode first (set to "true") so you can verify what will happen

Important notes:
- The `GITEA_ORGANIZATION` variable tells the tool where to create the repositories in Forgejo
- The organization must already exist in your Forgejo instance
- Your user token must have permission to create repositories in that organization
- Once you've verified everything looks good with dry-run, change `DRY_RUN="true"` to `DRY_RUN="false"` to perform the actual mirroring

If you want to preserve the organization structure from GitHub (i.e., keep the repositories in the same organization structure), you can add:

```
-e PRESERVE_ORG_STRUCTURE="true" \
```

If you only want to mirror the RetroDECK organization repositories and not your personal repositories, you can use:

```
-e ONLY_MIRROR_ORGS="true" \
-e INCLUDE_ORGS="RetroDECK" \
```

Full example with all options:

```bash
docker run -it --rm \
  -e GITHUB_USERNAME="your-github-username" \
  -e GITEA_URL="https://your-forgejo-instance-url" \
  -e GITEA_TOKEN="your-user-token-from-forgejo" \
  -e GITHUB_TOKEN="your-github-token" \
  -e GITEA_ORGANIZATION="RetroDECK" \
  -e MIRROR_ORGANIZATIONS="true" \
  -e ONLY_MIRROR_ORGS="true" \
  -e INCLUDE_ORGS="RetroDECK" \
  -e PRESERVE_ORG_STRUCTURE="true" \
  -e DRY_RUN="true" \
  arunavo4/mirror-to-gitea:latest
```

Let me know if you need any further assistance!

**Comment by jaedle** [comment]
Project is in maintenance mode, successor suggested.
